### DISCUSSION TITLE
[2023-07-10] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by wdanilo** [enso-org/enso#7247] [Weekly Demo Day]
## Design
This week I was mostly GUI design. This involved working on different aspects, including Component Browser, Visualizations, Cloud, Documentation, and expanded nodes.

## Component Browser
The design is finished and merged. Can be found [here](https://github.com/enso-org/design/blob/main/epics/component-browser/design2.md). The design can be also accessed in the [Enso Figma workspace](https://www.figma.com/files/project/31776253/Design?fuid=913498385423430287).

<img width="1439" alt="image" src="https://github.com/enso-org/design/assets/1623053/698879a2-aded-4e16-8fb5-42af61bb7ff9">

<img width="675" alt="image" src="https://github.com/enso-org/design/assets/1623053/d86fc819-1b7a-454d-8b09-026d04caae7b">

<img width="675" alt="image" src="https://github.com/enso-org/design/assets/1623053/0d74edbc-31b8-4d46-b3f5-30a217b9d3f2">

<br/>

## Other parts of GUI
After finishing the Component Browser design, we were chatting about other parts of the GUI that need to be consistent. These discussions spawned a lot of additional work which is needed earlier than later, as we are refining different parts of GUI before release. Moreover, leaving them for later date would cause way more discussions then now, as we would distract ourselves from one topic. Thus, we made the decision to focus on that and finish them in one go. There is 1-2 days left for finishing all the other parts of the design (including today). After that, I will finish the long overdue FRP PR.

**Comment by farmaazon** [comment]
Almost finished the new look of first CB element: the Component Grid aka Results List. Now it's single column, but changing the width of the main CB panel just looks fine.

https://github.com/enso-org/enso/assets/3919101/1d790861-b38d-4815-b5c5-5ebfbe518f26

Then I continued working on sorting the entries properly.

**Comment by vitvakatu** [comment]
Working on a redesign of the top bar: https://github.com/enso-org/enso/pull/7249
I accidentally broke the breadcrumbs' business logic, and I am fixing it now.

<img width="846" alt="Screenshot 2023-07-10 at 18 18 50" src="https://github.com/enso-org/enso/assets/6566674/3749ce51-9930-49cb-b18d-f97e447e06e7">

Also, visualizations resizing (presented last time) got merged yesterday: https://github.com/enso-org/enso/pull/7164. You should be able to resize every visualization now by dragging its bottom and right borders.

**Comment by mwu-tow** [comment]
* The Undo/Redo improvements PR has landed: #7209.
* The remote logging waits for tests/qa from cloud team: #6582 
* Updated VS version on the Windows CI runners (plus workarounds for Chocolatey issue);
* Started work on #7202. I looked into dashboard code and discussed with the Cloud team, there is already some infrastructure that I'll try to reuse.

**Comment by kazcw** [comment]
I've been working on fuzzy matching. The new matcher passes a test suite I wrote based on the specifications; there's one case remaining to handle in its support code. I will finish today and start on integrating the Enso Font.

**Comment by Frizi** [comment]
- Finishing first pass of big node style changes #6552
- moved and refactored graph editor layer stack away from hardcoded layers to graph-editor module.

<img width="690" alt="image" src="https://github.com/enso-org/enso/assets/919491/f0a3bc3a-099e-4c7b-aa57-328858dffda1">

**Comment by hubertp** [comment]
Some performance/IDE-related changes from backend:
- new benchmark API (#7101) that unifies our Java and Enso benchmarks
- support for file modifications notifications (#7239)
- moved a synchronous initialization step to be run in parallel (#7232)

**Comment by JanBeelte** [comment]
- Created a small NodeJS based file watcher tool: https://github.com/enso-org/enso/pull/7208

- Got the first multi-line Enso AI completion to work with my improved model.

Now working on integrating it into the demo UI and improving on stability.
